app/lib/features/space/pages/chats_page.dart

```diff
@@ -25,153 +26,184 @@ class SpaceChatsPage extends ConsumerWidget {
+      return Container();
+    }
+
+    final membership = ref.watch(roomMembershipProvider(spaceIdOrAlias));
+    bool canCreateSpace = membership.requireValue!.canString('CanLinkSpaces');
+
+    return Center(
+      heightFactor: 1,
+      child: EmptyState(
+        title: 'No chats in this space yet',
+        subtitle:
+            'Get the conversation going to start organizing collaborating',
+        image: 'assets/images/empty_chat.svg',
+        primaryButton: canCreateSpace
+            ? ElevatedButton(
+                onPressed: () => context.pushNamed(
+                  Routes.createChat.name,
+                  queryParameters: {'spaceId': spaceIdOrAlias},
+                  extra: 1,
+                ),
+                child: const Text('Create Space Chat'),
+              )
+            : null,
+      ),
+    );
+  }
+
+  Widget renderChats(BuildContext context, WidgetRef ref) {
+    final chats = ref.watch(relatedChatsProvider(spaceIdOrAlias));
+
+    return chats.when(
+      data: (rooms) {
+        return SliverAnimatedList(
+          initialItemCount: rooms.length,
+          itemBuilder: (context, index, animation) => SizeTransition(
+            sizeFactor: animation,
+            child: ConvoCard(
+              room: rooms[index],
+              showParent: false,
+              onTap: () => context.goNamed(
+                Routes.chatroom.name,
+                pathParameters: {'roomId': rooms[index].getRoomIdStr()},
               ),
             ),
-            error: (e, s) => SliverToBoxAdapter(
-              child: Text('Error loading related chats: $e'),
-            ),
-            loading: () =>
-                const SliverToBoxAdapter(child: Text('loading other chats')),
           ),
+        );
+      },
+      error: (error, stackTrace) => SliverToBoxAdapter(
+        child: Center(child: Text('Failed to load events due to $error')),
+      ),
+      loading: () => const SliverToBoxAdapter(
+        child: Skeletonizer(
+          child: ListTile(
+            title: Text('roomId'),
+            subtitle: Text('loading'),
+          ),
+        ),
+      ),
+    );
+  }
+
+  AsyncValue<
+      AutoDisposeStateNotifierProvider<FilteredSpaceHierarchyNotifier,
+          PagedState<Next?, SpaceHierarchyRoomInfo>>> relatedProvider(
+    WidgetRef ref,
+  ) {
+    return ref
+        .watch(spaceRelationsOverviewProvider(spaceIdOrAlias))
+        .whenData((spaces) => remoteChatHierarchyProvider(spaces));
+  }
+
+  Widget renderFurther(BuildContext context, WidgetRef ref) {
+    return relatedProvider(ref).when(
+      data: (provider) =>
+          RiverPagedBuilder<Next?, SpaceHierarchyRoomInfo>.autoDispose(
+        firstPageKey: const Next(isStart: true),
+        provider: provider,
+        itemBuilder: (context, item, index) => ConvoHierarchyCard(space: item),
+        noItemsFoundIndicatorBuilder: (context, controller) =>
+            _renderEmpty(context, ref),
```

Comment:
Just to confirm, `_renderEmpty` will show when there are no chat list which we have join as well as no related chat, correct?